nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -406,25 +406,23 @@ def nested(self, msg: str, attrs: Dict[str, str] = {}) -> _GeneratorContextManag
         return rootlog.nested(msg, my_attrs)
 
     def wait_for_monitor_prompt(self) -> str:
-        with self.nested("waiting for monitor prompt"):
-            assert self.monitor is not None
-            answer = ""
-            while True:
-                undecoded_answer = self.monitor.recv(1024)
-                if not undecoded_answer:
-                    break
-                answer += undecoded_answer.decode()
-                if answer.endswith("(qemu) "):
-                    break
-            return answer
+        assert self.monitor is not None
+        answer = ""
+        while True:
+            undecoded_answer = self.monitor.recv(1024)
+            if not undecoded_answer:
+                break
+            answer += undecoded_answer.decode()
```

Comment:
What if it's a three-byte character that starts at position 1023?
If all we have is `.decode()`, we can't decode anything until we have collected the whole stream.

Reply:
Note that all I've done in this commit is removing the `with self.nested` to decrease the log verbosity.
I think you need to git bisect/blame further.